src/commands/cmd_dispatcher.c

```diff
@@ -17,16 +17,31 @@
 typedef void(*Command_Handler)(void *args);
 
 // Read configuration flags, returning REDIS_MODULE_ERR if flag parsing failed.
-static int _read_flags(RedisModuleString **argv, int argc, bool *compact,
-					   long long *timeout, uint *graph_version, char **errmsg) {
+static int _read_flags
+(
+	RedisModuleString **argv,
+	int argc,
+	bool *compact,
+	long long *timeout,
+	bool *timeout_on_write,
+	uint *graph_version,
+	char **errmsg
+) {
 
 	ASSERT(compact);
 	ASSERT(timeout);
 
 	// set defaults
 	*compact = false;  // verbose
 	*graph_version = GRAPH_VERSION_MISSING;
-	Config_Option_get(Config_TIMEOUT, timeout);
+	Config_Option_get(Config_TIMEOUT_DEFAULT, timeout);
+	if(*timeout != CONFIG_TIMEOUT_NO_TIMEOUT) {
+		*timeout_on_write = true;
+	} else {
+		Config_Option_get(Config_TIMEOUT, timeout);
+		*timeout_on_write = false;
+	}
```

Comment:
All of the timeout logic here can be simplified once the comments on `config.c` are applied.